src/com/nicoles_change_maker/ChangeMaker.java

```diff
@@ -0,0 +1,61 @@
+package com.nicoles_change_maker;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
+/**
+ * Created by bears8yourface on 1/5/16.
+ */
+public class ChangeMaker {
+    double remainingDollars;
+
+    public HashMap makeChange(double dollarAmount) {
+        remainingDollars = dollarAmount;
+
+        if (checkDollarAmountIsValid(dollarAmount)) {
+            return evaluateEachCoinAmount();
+        }
+        throw new IllegalArgumentException();
+    }
+
+    private HashMap evaluateEachCoinAmount() {
+        HashMap changeCollection = new HashMap();
+        Iterator coinIterator = new CoinMap().getIterator();
+
+        while(coinIterator.hasNext()) {
```

Comment:
Modern Java has much nicer iteration abstractions, like the "enhanced for loop" (https://docs.oracle.com/javase/tutorial/java/nutsandbolts/for.html), which lets you avoid C-style `int i=0; i < collection.length(); i++` nonsense:

```java
for (Map.Entry coinValueMapping : coinIterator) {
  // do stuff with coinValueMapping
}
```

In this context, using an iterator seems a little off, since you have to call `.hasNext()` and `next()`, which are really implementation details. If you go with an enhanced for loop, you can just use an _`Iterable`_, like a `LinkedList` or `ArrayList`, e.g.:

```java
for (Entry valueMapping : coinMap.coinValues()) {
  evaluateCoinAmountForInclusionInChangeCollection(changeCollection, valueMapping);
}